packages/odyssey-react/src/components/FieldLabel/index.tsx

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React from 'react';
+import type {
+  FunctionComponent,
+  ReactText
+} from 'react';
+import { useCx } from '../../utils';
+
+export type FieldLabelVariant = 'label' | 'hint' | 'error' | 'optional';
+export type Props = {
+  /**
+   * The text content to be displayed to the user.
+   */
+  children: ReactText,
```

Comment:
Can we add another story/test to showcase a `FieldLabel` within a `FieldLabel`? I'd assume that would be how we'd support the following, where the "Optional" text is within the outer label component:


<img width="557" alt="Screen Shot 2021-06-17 at 9 49 31 PM" src="https://user-images.githubusercontent.com/20212241/122470516-0068f800-cf73-11eb-8910-febb70fd36ba.png">

Reply:
hmmm, yea I was trying to keep the storybook examples focused on just the component itself but there could be good value in showing it in context with other components too. I can add that in during the followups next sprint good thought @jmelberg-okta